packages/manager/apps/container/src/components/AgreementsUpdateModal/AgreementsUpdateModal.component.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  ODS_THEME_COLOR_HUE,
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import {
+  OsdsButton,
+  OsdsLink,
+  OsdsModal,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { OdsHTMLAnchorElementTarget } from '@ovhcloud/ods-common-core';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_TEXT_LEVEL,
+} from '@ovhcloud/ods-components';
+import { useAgreementsUpdate } from '@/hooks/agreements/useAgreements';
+import ApplicationContext from '@/context';
+import ovhCloudLogo from '@/assets/images/logo-ovhcloud.png';
+
+/*
+ Lifecycle management:
+  - If user is on the agreements page, we will not display the modal and let the page notify for modal change
+  once the user accept non-validated agreements or leave the page
+  - Wait until all necessary data (IAM authorization, non-validated agreements list) are loaded
+  - Once we have the data, check if they allow the display of the modal (IAM authorized + at least one non-validated
+  agreement), if the conditions are met, we show the modal, otherwise we switch to the next one
```

Comment:
```suggestion
  - Once we have the data, check if they allow the display of the modal (IAM authorized + at least one non-validated agreement), if the conditions are met, we show the modal, otherwise we switch to the next one
```